devicereg/devicereg.proto

```diff
@@ -75,18 +113,23 @@ message ClaimDeviceRequest {
 }
 
 // ClaimDeviceResponse is the message returned after successfully claiming a
-// device. We return here a key pair for the user, as well as a public key for
-// the device. The corresponding private key is used within the stream encoder
-// in order to encrypt data for the device.
+// device. We return here a key pair for the user, as well as the key pair for
+// the device. the device. The private key is used within the stream encoder in
+// order to encrypt data for the device for specified recipients, but we also
+// require the client to present the private key in order to apply new
+// entitlements to the device.
 message ClaimDeviceResponse {
   // The private part of a key pair for the individual user.
   string user_private_key = 1;
 
   // The public part of a key pair representing the individual user.
   string user_public_key = 2;
 
-  // The public key for the device (TODO - is this useful for any reason?)
+  // The public part of a key pair applied to the device.
   string device_public_key = 3;
+
+  // The private part of a key pair applied to the device.
+  string device_private_key = 4;
```

Comment:
Out of interest: am I correct in understanding that the Stream encoder doesn't need to store the owner private key?